frontend/src/components/FileBrowser/FileBrowser.js

```diff
@@ -40,34 +40,35 @@ export class FileBrowser extends React.Component<
         this.state = {
             currentWorkingDirectory: '',
             fileBrowserData: {},
-            isVisible: false,
+            isVisible: !this.props.startCollapsed,
         };
     }
 
-    componentWillReceiveProps(nextProps) {
-        if (nextProps.isRunBundleUIVisible === false && this.state.isVisible) {
-            this.setState({ isVisible: false });
-            this.getDOMNode()
-                .getElementsByClassName('file-browser-arrow')[0]
-                .click();
+    componentDidMount() {
+        if (!this.props.startCollapsed) {
+            this.updateFileBrowser('');
+        }
+    }
+
+    static getDerivedStateFromProps(nextProps, prevState) {
```

Comment:
I'm assuming you referred to this article for the replacement? https://medium.com/hackernoon/replacing-componentwillreceiveprops-with-getderivedstatefromprops-c3956f7ce607
just adding comment here in case in the future someone needs context

Reply:
Actually I referred to the official doc: https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html 
But thanks for putting the reference here!